### ISSUE TITLE
Ankidroid JS API not working

### ISSUE THREAD

**Issue by Toukir173** [ankidroid/Anki-Android#15437]
I was using API like to show the new card number on the top of my flashcards:

//Watermark with new card number
var newCardCount = parseInt(AnkiDroidJS.ankiGetNewCardCount(), 10);

document.getElementById("newCardCount").innerHTML = newCardCount;

But, since two days for some reason, AnkiDroidJS.ankiGetNewCardCount() like APIs are not working. Is there any issue or just the APIs have been changed that I don't know? Please help.

**Comment by david-allison** [comment]
What's the full code that you're using?

**Comment by david-allison** [comment]
* Possibly related: https://github.com/ankidroid/Anki-Android/issues/14953

**Comment by mikehardy** [comment]
Hi there! You have not included the version number you are using - I am going to make the bold (and maybe wrong!) assumption that you are enrolled in the beta program, and you were on 2.16.5 but our new 2.17beta1 and beta2 just hit the store and it may have incompatible JS API (since it is still a pretty new feature, we do break it incompatibility - apologies)


Following David's link this is the exact thing I think you need? https://github.com/ankidroid/Anki-Android/issues/14953#issuecomment-1851511564

**Comment by Toukir173** [comment]
> What's the full code that you're using?

<script>var n = 7;</script>

<div class="card" style="min-height: 100vh;" onclick="showAnswer()">

<p class="newCard" id="newCardCount" onclick="black()"></p>
<B><p id="Question"></p></B>
<p class="option"><br></p>

<br><br><br><br>

</div>





<script>
var InputAll="{{Q & A}}";
var Input=InputAll.split("</div><div><div><br><br>");
if(Input.length == 1){
Input=InputAll.split("</div><div><div>");
}
var Q = Input[n-1].split("ক) ");
sessionStorage.setItem('Q', JSON.stringify(Q));
document.getElementById("Question").innerHTML=Q[0];

O1 = Q[1].split("খ) ");
O2 = O1[1].split("গ) ");
O3 = O2[1].split("ঘ) ");
O4 = O3[1].split("ঙ) ");
O5 = "nan";
if (O4.length=="1"){
O4 = O4[0].split("<table><tbody><tr><td>Hide</td><td>ব্যাখ্যা</td><td>ফেভারিট থেকে মুছে ফেলুন</td></tr></tbody></table><div>");
} else {
O5 = O4[1].split("<table><tbody><tr><td>Hide</td><td>ব্যাখ্যা</td><td>ফেভারিট থেকে মুছে ফেলুন</td></tr></tbody></table><div>");
O5[0] = O5[0].endsWith("<br>") ? O5[0].slice(0, -4) : O5[0];
};

O1[0] = O1[0].endsWith("<br>") ? O1[0].slice(0, -4) : O1[0];
O2[0] = O2[0].endsWith("<br>") ? O2[0].slice(0, -4) : O2[0];
O3[0] = O3[0].endsWith("<br>") ? O3[0].slice(0, -4) : O3[0];
O4[0] = O4[0].endsWith("<br>") ? O4[0].slice(0, -4) : O4[0];

var Answer
if (Q.length=="3"){Answer = O1[0];};
if (O1.length=="3"){Answer = O2[0];};
if (O2.length=="3"){Answer = O3[0];};
if (O3.length=="3"){Answer = O4[0];};
if (O4.length=="3"){Answer = O5[0];};
sessionStorage.setItem('Ans',(Answer));

if (1 || Q[0].includes('****')) {
  Q[0] = Q[0].replace('****', ''); // Remove '****'
  Q[0] += `<span style='font-size: 12px'>${O1[0]}</span><span style='font-size: 12px'>${O2[0]}</span><span style='font-size: 12px'>${O3[0]}</span><span style='font-size: 12px'>${O4[0]}</span>`;
} else if (Q[0].includes('***')) {
Q[0] = Q[0].replace('***', '').replace('কোনটি', 'কোনগুলো').replace('is not', 'are').replace(' a ', ' ').replace(' নয়', '');
}
if (O5 != "nan") {
Q[0] += `<span style='font-size: 12px'>${O5[0]}</span>`;
}
document.getElementById("Question").innerHTML=Q[0];
//***End of Main Code


//Watermark with new card number
var newCardCount = parseInt(AnkiDroidJS.ankiGetNewCardCount(), 10);
var lrnCardCount = parseInt(AnkiDroidJS.ankiGetLrnCardCount(), 10);
var revCardCount = parseInt(AnkiDroidJS.ankiGetRevCardCount(), 10);
document.getElementById("newCardCount").innerHTML = newCardCount + lrnCardCount + revCardCount;


//Night Mode Color Correction
var mode = AnkiDroidJS.ankiIsInNightMode();
if(mode !== true){
		var cssStyleA = 'Ans';
		var cssStyleO = 'Others';
	} else {
		var cssStyleA = 'AnsBlck';
		var cssStyleO = 'OthersBlck';
	};
sessionStorage.setItem('cssStyleA',(cssStyleA));
var clr = 'green';
function chk(id){
if(Answer==Option[id-1]){	
var el = document.getElementById(id);
el.className = "Ans";
sessionStorage.setItem('clr',(clr));
	showAnswer();
 study.drawAnswer();
} else {	
var el = document.getElementById(id);
el.className = "Others";
clr = 'red';
};
sessionStorage.setItem('clr',(clr));
};;

</script>

**Comment by Toukir173** [comment]
> Hello! 👋 Thanks for logging this issue. Please remember we are all volunteers here, so some patience may be required before we can get to the issue. Also remember that the fastest way to get resolution on an issue is to propose a change directly, https://github.com/ankidroid/Anki-Android/wiki/Contributing

Thank you so much 😊

**Comment by Toukir173** [comment]
> * Possibly related: [Docs: JavaScript APIs documentation needs updating #14953](https://github.com/ankidroid/Anki-Android/issues/14953)

Thank you so much 😊

**Comment by david-allison** [comment]
Also:

```
<script>
    var jsApi = { "version": "0.0.2", "developer": your email here };
    var api = AnkiDroidJS.init(jsApi);
    var response = api.FUNCTION(args).then(() => {});
</script>
```
